### PR TITLE
SMP: Hide stats the never finish loading

### LINKED ISSUE #75072: Sites page: stats never finish loading for site without stats
### Quick summary

We display a small chart showing stats on the Sites page when it's displayed in the table view. For sites that don't have stats enabled, we always display a loading indicator.

Screenshot 2023-03-29 at 14 16 28

### Steps to reproduce

1. Create a site that doesn't support stats e.g. eCommerce Trial
2. Load the Sites page and check the site on the list

### What you expected to happen

There is no stats chart.

### What actually happened

There is a stats loading indicator that always remains visible.

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

Request to `/rest/v1.1/sites/$blog_id/stats/insights` endpoint fails with 404 and "This blog does not have the Stats module enabled" message.

### PR DESCRIPTION
**Pull request by katinthehatsite**
* Closes: https://github.com/Automattic/wp-calypso/issues/75072

## Proposed Changes

* This PR replaces the infinitely loading bar with the "Stats Off"  on the SMP page for sites that don't have stats enabled (some staging, Woo Express, and self-hosted sites on which the request to retrieve stats fails). 

## Testing Instructions

* Create a site with a `Woo Express: Trial (Monthly)` plan (you can add it to a site from Store Admin)
* Once the site is created, navigate to `https://wordpress.com/sites`. Make sure to select the table view
* Find the your `Woo Express: Trial` site and observe the **Stats** column
* Check that the infinitely loading stats bar is replaced with "Stats Off" text

Screenshot 2023-05-16 at 4 06 53 PM

* Hover over "Stats Off"
* Observe the popover appearing that says "Stats are disabled for this site"

Screenshot 2023-05-16 at 4 07 46 PM

* Hover away from "Stats Off"
* Observe the popover going away
* This view is disabled on mobile but is visible on some tablets. You can test with Chrome dev tools with iPad Air:

Screenshot 2023-05-16 at 4 13 11 PM

* Switch your account language from `me/account` to RTL (like Arabic) and check that the `Stats Off` indicator works as expected:

Screenshot 2023-05-16 at 4 17 01 PM

Screenshot 2023-05-16 at 4 18 50 PM

### Some Notes

* While working on this PR, I discovered that it is possible to enable the stats for Woo Express sites from this page `https://wordpress.com/stats/day/{yourecommercetrialsite}` and they appear under **Jetpack Stats**:

Screenshot 2023-05-16 at 3 23 16 PM

Once you activate them once from that page, they will be loaded on the SMP page (`https://wordpress.com/sites`) similar to all other sites and the loading bar will go away. I am not sure if it is intentional to allow stats on the trial site. If it is, should we link from the popover text "Stats are disabled for this site" to `https://wordpress.com/stats/day/` where the stats can be enabled?

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/sites-dashboard/components/sites-table-row.tsx [hunk 1]
```diff
@@ -78,6 +80,59 @@ const ListTileSubtitle = styled.div`
...
+			onMouseOut={ handleOnMouseExit }
+			onFocus={ handleOnMouseEnter }
+			onBlur={ handleOnMouseExit }
+		>
+			<StatsOffIndicatorStyled className="tooltip-target" ref={ tooltipRef }>
+				{ translate( 'Stats Off' ) }
+			</StatsOffIndicatorStyled>
+			{ showPopover && (
+				<Popover isVisible={ true } context={ tooltipRef.current }>
+					<PopoverContent>{ translate( 'Stats are disabled on this site' ) }</PopoverContent>
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-table-row.tsx]:
```suggestion
					<PopoverContent>{ translate( 'Stats are disabled on this site.' ) }</PopoverContent>
```

File: client/sites-dashboard/components/sites-table-row.tsx [hunk 2]
```diff
@@ -78,6 +80,59 @@ const ListTileSubtitle = styled.div`
...
+	border-radius: 4px;
+	background-color: var( --studio-gray-5 );
+	color: var( --studio-gray-100 );
+	font-size: 12px;
+	padding: 0 7px;
+	display: inline-flex;
+`;
+
+const PopoverContent = styled.div`
+	font-size: 14px;
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-table-row.tsx:95]: This is a slightly larger font than the Media Library tooltip:

image

The menu items are at 13 px too:

image

Should we go a little smaller? @javierarce ?

image

File: client/sites-dashboard/components/sites-table-row.tsx [hunk 3]
```diff
@@ -78,6 +80,59 @@ const ListTileSubtitle = styled.div`
...
+	const handleOnMouseExit = () => {
+		setShowPopover( false );
+	};
+
+	return (
+		<div
+			onMouseOver={ handleOnMouseEnter }
+			onMouseOut={ handleOnMouseExit }
+			onFocus={ handleOnMouseEnter }
+			onBlur={ handleOnMouseExit }
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-table-row.tsx:122]: I was curious whether there was a better way to trigger the popover.

Here are a couple examples:

https://github.com/Automattic/wp-calypso/blob/c6cf4578a2353ff0e165c8e50f206fbad1fda8e0/client/components/info-popover/index.jsx#L123-L135

https://github.com/Automattic/wp-calypso/blob/c6cf4578a2353ff0e165c8e50f206fbad1fda8e0/client/blocks/plan-storage/tooltip.jsx#L9-L22

It seems like it's generally the responsibility of the caller, and there isn't a standard pattern to use. Your approach seems fine.

**Review by danielbachhuber — APPROVED** [review]
Works as expected 👍 

All of my feedback is non-blocking.

File: client/sites-dashboard/components/sites-table-row.tsx (hunk 2 above)
Comment (javierarce) [reply client/sites-dashboard/components/sites-table-row.tsx:95]: No, let's use the same style as the [InfoPopover component](https://wpcalypso.wordpress.com/devdocs/design/info-popover).

```
color: var(--color-neutral-50);
font-size: .875rem;
max-width: 220px;
padding: 16px;
text-align: left;
```

So the only things we should change are the color and the padding.

**Comment by javierarce** [comment]
Great work, @katinthehatsite! 

Besides my comment [here](https://github.com/Automattic/wp-calypso/pull/76980#discussion_r1196313043), I just saw to small things:


- If it's not too difficult to do, it would also be nice to add a tiny margin between the tag and the popover. 
- And I think we are not using title case for the badges, so let's say `Stats off` instead of `Stats Off`.

File: client/sites-dashboard/components/sites-table-row.tsx (hunk 3 above)
Comment (katinthehatsite) [reply client/sites-dashboard/components/sites-table-row.tsx:122]: Yeah I looked into a few options but could not find a common approach/ pattern around how it was done before so decided on this one 😅

**Comment by katinthehatsite** [comment]
I added the margin as requested and made small fixes based on the feedback

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8006739

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8006739).

**Pull request merged**